spec/std/hash_spec.cr

```diff
@@ -257,6 +257,15 @@ describe "Hash" do
       a = {1 => 2}
       a.delete(2).should be_nil
     end
+
+    describe "with block" do
+      it "executes block if key is not found" do
+        found = true
+        a = {1 => 2}
+        a.delete(3) { found = false }
+        found.should be_false
+      end
+    end
```

Comment:
You should also spec that the value of the block is returned. And that the method still finds a value if it exists.